hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequestWithFSO.java

```diff
@@ -172,7 +171,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
             omPathInfo.getLeafNodeName(),
             keyArgs, omPathInfo.getLeafNodeObjectId(),
             omPathInfo.getLastKnownParentId(), trxnLogIndex,
-            OzoneAclUtil.fromProtobuf(keyArgs.getAclsList()));
+            inheritDefaultAcls(keyArgs, omBucketInfo.getAcls()));
```

Comment:
If ancestor directory exists, the new directories should inherit from that ancestor instead of the bucket.  I think `omPathInfo` has the information about the ancestor or bucket in both cases.

Reply:
> If ancestor directory exists, the new directories should inherit from that ancestor instead of the bucket. I think `omPathInfo` has the information about the ancestor or bucket in both cases.

@adoroszlai  Thanks for suggestion. I have a little doubt. If ancestor directory exists, will `both DEFAULT and ACCESS ACLs` be inherited, or only the `DEFAULT acl` will be inherited?

Reply:
Per the [design doc](https://issues.apache.org/jira/secure/attachment/12997587/Design%20Doc-%20Native%20ACL%20support%20for%20Ozone.pdf) for ACLs in Ozone, inheritance is only supported on default ACLs.  I don't know if actual implementation follows that.